source/mainnet/smart-contracts/tutorials/piggy-bank/deploying.rst

```diff
@@ -0,0 +1,160 @@
+=======================================
+
+This is the fourth :ref:`part of a tutorial<piggy-bank>` on smart contract
+development.
+So far you have written and tested a piggy bank smart contract in the Rust_ programming language.
+
+Furthermore, you set up your testnet node on a server, created an account that is funded with some CCD, and  imported your account key into the ``concordium-client`` configuration.
+
+This part will focus on how you can deploy your developed piggy bank smart contract to the Concordium testnet and interact with it.
+
+
+.. warning::
+
+   The reader is assumed to have basic knowledge of what a blockchain and smart
+   contract is, and some experience with Rust_.
+
+
+Preparation
+===========
+
+Before you start, make sure you completed the previous parts of the tutorial and have a running testnet node that has caught up to the head of the blockchain and an account with some CCD setup.
+
+Building a wasm module
+======================
+
+You have already built a wasm module from your piggy bank smart contract in a :ref:`previous part<cargo-concordium-build>` of this tutorial.
+
+Let's repeat this step since you might have done some changes to the smart contract code in :ref:`part 2<piggy-bank-testing>` of this tutorial. Move to the folder where you saved your piggy bank smart contract and use ``cargo concordium build`` to create a wasm module.
+
+.. code-block:: console
+
+   $cargo concordium build
+
+.. Note::
+   This command creates a module ``piggy_bank_part2.wasm.v1`` in the folder ``./target/concordium/wasm32-unknown-unknown/release/``.
+   Please use the wasm module file suffixed with ``v1`` when deploying to the chain. The ``piggy_bank_part2.wasm`` file (without the ``v1`` suffix) is not the ``v0`` module, but rather the raw wasm module produced by cargo.
+   The ``v1`` suffixed module has been stripped for debugging information, which makes it significantly smaller. Modules already include the smart contract version (and the size of the module).
+
+
+The ``piggy_bank_part2.wasm.v1`` file can be deployed to the Concordium chain. Let’s transfer the ``piggy_bank_part2.wasm.v1`` file to the place where you are running your concordium-client tool. Move to the folder as well.
+
+Deploying a module
+==================
+You are now set up to deploy the piggy bank module to the testnet chain. Let's give the module the name ``piggy_bank_part2_module`` so it can be referenced in this tutorial.
+
+.. Note::
+   Please remember to adjust the following commands when you are running your ``concordium-client`` tool locally and want to connect to your node on the server. You can look up the different options on how to connect the ``concordium-client`` tool to your testnet node :ref:`here<interacting-with-your-testnet-node>`.
+
+.. code-block:: console
+
+   $./concordium-client module deploy piggy_bank_part2.wasm.v1 --sender <account-name> --name piggy_bank_part2_module --grpc-port 10001
```

Comment:
We can mention that the names are only stored locally in the concordium client configuration. They are not visible to others on the chain. So they only exist as a convenience to the user 😊 
(That is the case for all the names/aliases, i.e., contract, module, and account names)